src/app/content/practiceQuestions/components/Filters.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import ChapterFilter from '../../components/popUp/ChapterFilter';
+import Filters, { FilterDropdown, FiltersTopBar } from '../../components/popUp/Filters';
+import { isLinkedArchiveTreeSection } from '../../guards';
+import * as contentSelectors from '../../selectors';
+import { findArchiveTreeNodeById } from '../../utils/archiveTreeUtils';
+import { setSelectedSection } from '../actions';
+import * as selectors from '../selectors';
+
+export default () => {
+  const [open, setOpen] = React.useState(false);
+  const locationFilters = useSelector(selectors.practiceQuestionsLocationFilters);
+  const selectedSection = useSelector(selectors.selectedSection);
+  const book = useSelector(contentSelectors.book);
+  const dispatch = useDispatch();
+  const setFilters = React.useCallback(({ locationIds }: { locationIds: string[] }) => {
+    const clickedSectionId = locationIds.pop();
+    if (!clickedSectionId) {
+      // user clicked on the selected section
+      setOpen(false);
+      return;
+    }
+    const search = book && locationIds.length
+      ? findArchiveTreeNodeById(book.tree, clickedSectionId)
+      : null;
+    const section = search && isLinkedArchiveTreeSection(search) ? search : null;
```

Comment:
you're doing a lot of work to find the section here, maybe we could change the I/O to work on the sections instead of just the ids? it would be more flexible

Reply:
I've started working on it and it was a rabbit hole... :D I'm almost done with it, but it required a lot of changes